File: Leetcode/POTD/Largest 3-Same-Digit Number in String.java
class Solution {
    public String largestGoodInteger(String num) {
        String result = "";
        for (int i = 0; i < num.length() - 2; i++) {
            if (num.charAt(i) == num.charAt(i+1) && num.charAt(i+1) == num.charAt(i+2)) {
                String current = num.substring(i, i+3);
                if (current.compareTo(result) > 0) {
                    result = current;
                }
            }
        }
        return result;
    }
}

File: Leetcode/POTD/Largest 3-Same-Digit Number in String.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        string fin = "";
        for(int i=0; i<num.size(); i++)
        {
            if(num[i]==num[i+1] && num[i+1] == num[i+2])
            {
                string s = num.substr(i,3);
                if(s>fin)   fin = s;
            }
        }
        return fin;
    }
};
